### ISSUE TITLE
Presets Panel: Nested Interactive Elements

### ISSUE THREAD

**Issue by derekjackson-das** [danny-avila/LibreChat#5489]
## Issue Description
Each preset is has the role of "option" but also contain more interactive items nested in the option to perform actions like delete or edit. This is not an expected pattern for the role, and many non-visual and low-vision users will not anticipate the nested actions in an "option" object.

- #5450 
- #5451 
- #5452 
- #5453 

## WCAG Criteria
[1.3.1 Info and Relationships - A](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)

## Notes
Separate the different functions of the task into their own interactive elements. The "option" role does not need to be used here because it is typically understood to be part of a `select` HTML element or a widget with a `listbox` role, which is not the case for the preset choices. 

In addition, the `option` element should only contain text, so including interactive elements becomes somewhat of an anti-pattern. 

One alternative is to have each preset grouped but put the actions into their own interactive elements, e.g., buttons. A very simple example of an alternative UI can be seen here: [CodePen Demo](https://cdpn.io/pen/debug/JoPwYZN). 

<img width="933" alt="Screenshot of the accessibility tree in Chrome's devtools showing each preset as a group and the associated actions within the group as individual buttons." src="https://github.com/user-attachments/assets/03eb1122-4d3f-4160-ba1e-1fd7a01731c0" />

More details about the option role: [MDN Option Role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/option_role).

## Screenshot or Screen Recording

<img width="1042" alt="Screen shot of the presets panel with one of the presets high-lighted and the corresponding code, listed in the Relevant Code below, with the role='options' attribute circled."  src="https://github.com/user-attachments/assets/f4cd54dc-dea8-421f-8db8-f550355ed475" />

## Relevant Code
```html
<div type="button">
  <div
    role="option"
    aria-selected="false"
    aria-label="New Preset:"
    data-testid="preset-item-[object Object]"
    class="group m-1.5 flex cursor-pointer gap-2 rounded px-5 py-2.5 !pr-3 text-sm !opacity-100 hover:bg-black/5 focus:ring-0 radix-disabled:pointer-events-none radix-disabled:opacity-50 dark:hover:bg-gray-600 md:min-w-[240px]"
    tabindex="0"  >
    ...
  </div>
</div>
